app/src/main/java/io/scalac/degree/android/adapter/schedule/model/TalksScheduleItem.java

```diff
@@ -20,54 +26,103 @@ public TalksScheduleItem(long startTime, long endTime, int strIndex, int stpInde
         otherSlots = new ArrayList<>();
     }
 
-    public void setFavouredSlots(List<SlotApiModel> slots) {
-        favouredSlots = new ArrayList<>(slots.size());
-        favouredSlots.addAll(slots);
+    public void addFavouredSlot(SlotApiModel model) {
+        favouredSlots.add(model);
+    }
+
+    public void addOtherSlot(SlotApiModel model) {
+        otherSlots.add(model);
     }
 
-    public void setOtherSlots(List<SlotApiModel> slots) {
-        otherSlots = new ArrayList<>(slots.size());
-        otherSlots.addAll(slots);
+    public int talksCount() {
+        return countTalks(otherSlots) + countTalks(favouredSlots);
+    }
+
+    public int tracksCount() {
+        return countTracks(otherSlots) + countTracks(favouredSlots);
     }
 
     @Override
     public int getSize() {
-        return favouredSlots.size() + otherSlots.size() + EXTRA_TIMESPAN_ELEMENT_COUNT;
+        final int result;
+        if (isOthersVisible) {
+            result = favouredSlots.size() + otherSlots.size();
+        } else {
+            result = favouredSlots.size();
+        }
+        return result + EXTRA_TIMESPAN_ELEMENT_COUNT
+                + EXTRA_OPEN_MORE_ELEMENT_COUNT;
     }
 
     @Override
     @ScheduleDayLineupAdapter.ViewType
     public int getItemType(int position) {
-
-        // TODO Handle breaks and favs!
-
-        final int probableTimespanIndex = position - getStartIndex();
-        if (probableTimespanIndex == TIMESPAN_INDEX) {
-            return ScheduleDayLineupAdapter.TIMESPAN_VIEW;
+        final int calendarOpenMoreIndexWithNoFavs = TIMESPAN_INDEX + 1;
+        final int calendarOpenMoreIndexWithFavs = TIMESPAN_INDEX + favouredSlots.size() + 1;
+
+        final int innerIndex = position - getStartIndex();
+
+        if (favouredSlots.isEmpty()) {
+            if (innerIndex == TIMESPAN_INDEX) {
```

Comment:
tego ifa na pewno mozesz przed `if (favouredSlots.isEmpty()) {`, ostatni warunek tez jest taki sam wiec zdaje sie mozna to jakos uproscic jeszcze bardziej